Authentication Link and IDP Fails with 400 Bad Request After Migrating to Version 26 and Delete Authentification

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

authentication

### Describe the bug

Hello

After upgrade Keycloak 19.0 to 26.0.5, all work correctly 

But after delete Authentification , admin portal displa error and logs keycloak 

Cannot invoke "org.keycloak.models.AuthenticatorConfigModel.getAlias()" because "config" is null

remove authentification in 19.0 don't create error

### Version

26.0.5

### Regression

- [ ] The issue is a regression

### Expected behavior

No error

### Actual behavior

Cannot invoke "org.keycloak.models.AuthenticatorConfigModel.getAlias()" because "config" is null

After delete one authentification

### How to Reproduce?

remove authentification

### Anything else?

_No response_

@rmartinc thanks

this is a full stack
```
2024-11-15 00:20:28,139 DEBUG [org.hibernate.orm.sql.exec] (executor-thread-4) Skipping reading Query result cache data: cache-enabled = false, cache-mode = NORMAL
2024-11-15 00:20:28,139 DEBUG [org.hibernate.SQL] (executor-thread-4) select ce1_0.ID from CLIENT ce1_0 left join CLIENT_AUTH_FLOW_BINDINGS afb1_0 on ce1_0.ID=afb1_0.CLIENT_ID where ce1_0.REALM_ID=? and afb1_0.BINDING_NAME=? and afb1_0.FLOW_ID=? order by ce1_0.CLIENT_ID offset ? rows fetch first ? rows only
2024-11-15 00:20:28,139 DEBUG [org.hibernate.orm.results] (executor-thread-4) Initializer list is empty
2024-11-15 00:20:28,140 DEBUG [org.keycloak.transaction.JtaTransactionWrapper] (executor-thread-4) JtaTransactionWrapper commit. Request Context: HTTP GET /admin/realms/microcks/ui-ext/authentication-management/flows
2024-11-15 00:20:28,140 DEBUG [org.hibernate.engine.transaction.internal.TransactionImpl] (executor-thread-4) On TransactionImpl creation, JpaCompliance#isJpaTransactionComplianceEnabled == false
2024-11-15 00:20:28,140 DEBUG [org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl] (executor-thread-4) Initiating JDBC connection release from beforeTransactionCompletion
2024-11-15 00:20:28,140 FINE [org.postgresql.jdbc.PgConnection] (executor-thread-4) setAutoCommit = true
2024-11-15 00:20:28,140 DEBUG [org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl] (executor-thread-4) Initiating JDBC connection release from afterTransaction
2024-11-15 00:20:28,140 DEBUG [org.keycloak.transaction.JtaTransactionWrapper] (executor-thread-4) JtaTransactionWrapper end. Request Context: HTTP GET /admin/realms/microcks/ui-ext/authentication-management/flows
2024-11-15 00:20:29,686 DEBUG [org.keycloak.transaction.JtaTransactionWrapper] (executor-thread-4) new JtaTransactionWrapper. Was existing transaction suspended: false Request Context: HTTP GET /admin/realms/microcks/authentication/flows
2024-11-15 00:20:29,687 DEBUG [org.keycloak.models.sessions.infinispan.changes.UserSessionPersistentChangelogBasedTransaction] (executor-thread-4) user-session found in cache for sessionId=f8c0721f-d69a-41e7-87b1-177a599fb163 offline=false 1731630027
2024-11-15 00:20:29,687 DEBUG [org.keycloak.models.sessions.infinispan.changes.ClientSessionPersistentChangelogBasedTransaction] (executor-thread-4) client-session found in cache for sessionId=a2a0620a-0310-3706-bcb2-6eda763382b9, offline=false
2024-11-15 00:20:29,687 DEBUG [org.keycloak.services.resources.admin.AdminRoot] (executor-thread-4) authenticated admin access for: admin
2024-11-15 00:20:29,688 DEBUG [org.keycloak.models.sessions.infinispan.PersistentUserSessionProvider] (executor-thread-4) getUserSessionWithPredicate(f8c0721f-d69a-41e7-87b1-177a599fb163): found in local cache
2024-11-15 00:20:29,690 DEBUG [org.keycloak.transaction.JtaTransactionWrapper] (executor-thread-4) JtaTransactionWrapper commit. Request Context: HTTP GET /admin/realms/microcks/authentication/flows
2024-11-15 00:20:29,690 DEBUG [org.keycloak.transaction.JtaTransactionWrapper] (executor-thread-4) JtaTransactionWrapper end. Request Context: HTTP GET /admin/realms/microcks/authentication/flows
2024-11-15 00:20:29,690 DEBUG [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-4) Error response Bad Request: com.fasterxml.jackson.databind.JsonMappingException: Cannot invoke "org.keycloak.models.AuthenticatorConfigModel.getAlias()" because "config" is null
at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._wrapAsIOE(DefaultSerializerProvider.java:531)
at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:504)
at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:341)
at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize(ObjectWriter.java:1574)
at com.fasterxml.jackson.databind.ObjectWriter._writeValueAndClose(ObjectWriter.java:1275)
at com.fasterxml.jackson.databind.ObjectWriter.writeValue(ObjectWriter.java:1098)
at io.quarkus.resteasy.reactive.jackson.runtime.serialisers.FullyFeaturedServerJacksonMessageBodyWriter.writeResponse(FullyFeaturedServerJacksonMessageBodyWriter.java:80)
at org.jboss.resteasy.reactive.server.core.ServerSerialisers.invokeWriter(ServerSerialisers.java:217)
t org.jboss.resteasy.reactive.server.core.ServerSerialisers.invokeWriter(ServerSerialisers.java:185)
at org.jboss.resteasy.reactive.server.core.serialization.FixedEntityWriter.write(FixedEntityWriter.java:28)
at org.jboss.resteasy.reactive.server.handlers.ResponseWriterHandler.handle(ResponseWriterHandler.java:34)
at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:147)
at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NullPointerException: Cannot invoke "org.keycloak.models.AuthenticatorConfigModel.getAlias()" because "config" is null
at org.keycloak.models.utils.ModelToRepresentation.toRepresentation(ModelToRepresentation.java:931)
at org.keycloak.models.utils.ModelToRepresentation.lambda$toRepresentation$8(ModelToRepresentation.java:923)
at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
at java.base/java.util.LinkedList$LLSpliterator.forEachRemaining(LinkedList.java:1249)
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
at org.keycloak.models.utils.ModelToRepresentation.toRepresentation(ModelToRepresentation.java:923)
at org.keycloak.services.resources.admin.AuthenticationManagementResource.lambda$getFlows$3(AuthenticationManagementResource.java:230)
at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
at java.base/java.util.stream.ReferencePipeline.forEachOrdered(ReferencePipeline.java:601)
at com.fasterxml.jackson.datatype.jdk8.StreamSerializer.serialize(StreamSerializer.java:71)
at com.fasterxml.jackson.datatype.jdk8.StreamSerializer.serialize(StreamSerializer.java:15)
at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:502)
... 19 more
```

--- a/server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java
+++ b/server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java
@@ -930,7 +930,9 @@ public class ModelToRepresentation {
         AuthenticationExecutionExportRepresentation rep = new AuthenticationExecutionExportRepresentation();
         if (model.getAuthenticatorConfig() != null) {
             AuthenticatorConfigModel config = new DeployedConfigurationsManager(session).getAuthenticatorConfig(realm, model.getAuthenticatorConfig());
-            rep.setAuthenticatorConfig(config.getAlias());
+            if (config != null) {
+                rep.setAuthenticatorConfig(config.getAlias());
+            }
         }
         rep.setAuthenticator(model.getAuthenticator());
         rep.setAuthenticatorFlow(model.isAuthenticatorFlow());

--- a/services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java
+++ b/services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java
@@ -663,6 +663,19 @@ public class AuthenticationManagementResource {
         return result;
     }
 
+    private String getAuthenticationConfig(String flowAlias, AuthenticationExecutionModel model) {
+        if (model.getAuthenticatorConfig() == null) {
+            return null;
+        }
+        AuthenticatorConfigModel config = new DeployedConfigurationsManager(session).getAuthenticatorConfig(realm, model.getAuthenticatorConfig());
+        if (config == null) {
+            logger.warnf("Authenticator configuration '%s' is missing for execution '%s' (%s) in flow '%s'",
+                    model.getAuthenticatorConfig(), model.getId(), model.getAuthenticator(), flowAlias);
+            return null;
+        }
+        return config.getId();
+    }
+
     public void recurseExecutions(AuthenticationFlowModel flow, List<AuthenticationExecutionInfoRepresentation> result, int level) {
         AtomicInteger index = new AtomicInteger(0);
         realm.getAuthenticationExecutionsStream(flow.getId()).forEachOrdered(execution -> {
@@ -682,7 +695,7 @@ public class AuthenticationManagementResource {
                     rep.getRequirementChoices().add(AuthenticationExecutionModel.Requirement.REQUIRED.name());
                     rep.getRequirementChoices().add(AuthenticationExecutionModel.Requirement.DISABLED.name());
                     rep.setProviderId(execution.getAuthenticator());
-                    rep.setAuthenticationConfig(execution.getAuthenticatorConfig());
+                    rep.setAuthenticationConfig(getAuthenticationConfig(flow.getAlias(), execution));
                 } else if (AuthenticationFlow.CLIENT_FLOW.equals(flowRef.getProviderId())) {
                     rep.getRequirementChoices().add(AuthenticationExecutionModel.Requirement.ALTERNATIVE.name());
                     rep.getRequirementChoices().add(AuthenticationExecutionModel.Requirement.REQUIRED.name());
@@ -732,7 +745,7 @@ public class AuthenticationManagementResource {
                 }
 
                 rep.setProviderId(providerId);
-                rep.setAuthenticationConfig(execution.getAuthenticatorConfig());
+                rep.setAuthenticationConfig(getAuthenticationConfig(flow.getAlias(), execution));
                 result.add(rep);
             }
         });

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/authentication/AuthenticatorConfigTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/authentication/AuthenticatorConfigTest.java
@@ -23,6 +23,9 @@ import org.keycloak.authentication.authenticators.broker.IdpCreateUserIfUniqueAu
 import org.keycloak.authentication.authenticators.broker.IdpDetectExistingBrokerUserAuthenticatorFactory;
 import org.keycloak.events.admin.OperationType;
 import org.keycloak.events.admin.ResourceType;
+import org.keycloak.models.AuthenticatorConfigModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.representations.idm.AuthenticationExecutionExportRepresentation;
 import org.keycloak.representations.idm.AuthenticationExecutionInfoRepresentation;
 import org.keycloak.representations.idm.AuthenticationFlowRepresentation;
 import org.keycloak.representations.idm.AuthenticatorConfigInfoRepresentation;
@@ -219,6 +222,36 @@ public class AuthenticatorConfigTest extends AbstractAuthenticationTest {
         Assert.assertEquals(404, nfe.getResponse().getStatus());
         nfe = Assert.assertThrows(NotFoundException.class, () -> authMgmtResource.getAuthenticatorConfig(config2Id));
         Assert.assertEquals(404, nfe.getResponse().getStatus());
+    }
+
+    @Test
+    public void testMissingConfig() {
+        AuthenticatorConfigRepresentation cfg = newConfig("foo", IdpCreateUserIfUniqueAuthenticatorFactory.REQUIRE_PASSWORD_UPDATE_AFTER_REGISTRATION, "true");
+        final String cfgId = createConfig(executionId, cfg);
+        final String realmId = testRealmId;
+        AuthenticatorConfigRepresentation cfgRep = authMgmtResource.getAuthenticatorConfig(cfgId);
+        Assert.assertNotNull(cfgRep);
+
+        testingClient.server().run(session -> {
+            // emulating a broken config id, remove the config but do not remove the link in the authenticator
+            RealmModel realm = session.realms().getRealm(realmId);
+            AuthenticatorConfigModel config = realm.getAuthenticatorConfigById(cfgId);
+            realm.removeAuthenticatorConfig(config);
+        });
+
+        // check the flow can be read and execution has no config
+        AuthenticationFlowRepresentation flow = authMgmtResource.getFlow(flowId);
+        AuthenticationExecutionExportRepresentation execExport = flow.getAuthenticationExecutions().stream()
+                .filter(ae -> IdpCreateUserIfUniqueAuthenticatorFactory.PROVIDER_ID.equals(ae.getAuthenticator()))
+                .findAny()
+                .orElse(null);
+        Assert.assertNotNull(execExport);
+        Assert.assertNull(execExport.getAuthenticatorConfig());
+
+        // check the execution can be read with no configuration assigned
+        AuthenticationExecutionInfoRepresentation execInfo = findExecutionByProvider(
+                IdpCreateUserIfUniqueAuthenticatorFactory.PROVIDER_ID, authMgmtResource.getExecutions("firstBrokerLogin2"));
+        Assert.assertNull(execInfo.getAuthenticationConfig());
     }
 
     private String createConfig(String executionId, AuthenticatorConfigRepresentation cfg) {

